.github/workflows/ci.yml

```diff
@@ -123,6 +121,51 @@ jobs:
       - name: Verible FPV
         run: ./ci/scripts/verible-lint.sh fpv
 
+  build_docs:
+    name: Build documentation
+    runs-on: ubuntu-22.04
+    needs: quick_lint
+    env:
+      BUCKET: gold-hybrid-255313-prod
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Bitstream cache requires all commits.
+      - name: Prepare environment
+        uses: ./.github/actions/prepare-env
+        with:
+          service_account_json: '${{ secrets.BAZEL_CACHE_CREDS }}'
+      - name: Build documentation
+        run: util/site/build-docs.sh build
+      - uses: google-github-actions/setup-gcloud@v2
```

Comment:
I think this conditional section where we do the upload only for `master` pushes is quite complex and would benefit from moving to a composite action. What do you think?

Reply:
I mean the whole thing is quite complex. It's 28 lines with those four mixed in that have to be kept in sync.

Reply:
What do you mean by kept in sync?

Reply:
When this is backported all four `master`s need to be changed

Reply:
How about:
* Make setup-gcloud be part of prepare-env and unconditionally executed for non-pull_request triggers (a couple of seconds, but only for pushes)
* Merge all bash steps into 1 so there's only single line with `master` as condition.

When backporting we should change the action on master to account for branch name and then backport as a whole.

Reply:
Sure, that sounds fine